CWE-94-Python_darwin/mit-han-lab_neurips-micronet/u.py

```
@@ -0,0 +1,944 @@
+
+    def load_yaml(self):
+        with open(self, 'r') as f:
+            return yaml.safe_load(f)
+
+    def save_yaml(self, obj):
+        obj = recurse(obj, lambda x: x._ if type(x) is Path else dict(x) if type(x) is dict else x)
+        with open(self, 'w') as f:
+            yaml.dump(obj, f, default_flow_style=False, allow_unicode=True)
+
+    def load(self):
+        return eval('self.load_%s' % self._ext)()
```

Ruff (0.13.2)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)